decisions.md

```diff
@@ -0,0 +1,91 @@
+
+### How is the NF script structured?
+
+The NF script contains the following structures:
+#### 2.1 DSL and Parameters
+#### 2.2 Definition of processes
+#### 2.3 Definition of workflows
+#### 2.4 Main workflow
+
+Check this source code example: [seq_ocrd_wf_many.nf](https://github.com/MehmedGIT/OPERANDI_TestRepo/blob/master/ExampleWorkflows/Nextflow/workflow4/seq_ocrd_wf_many.nf)
+
+TODO: I will provide more structure-related details here based on the example above.
+
+### Which features of NF do we use, i.e. what features have to be implemented in potential implementations?
+The minimally used features for local runs are the parameters, processes, process decorators, and workflows. I will provide further answers to any following questions related to this main question. I am not sure what else to cover here for now.
+
+## How does parallelization work, both within works and across works?
+
+A Nextflow workflow script contains several processes. Processes are executed independently and are isolated from each other (i.e. they do not have a shared memory space). Communication between the processes is possible only through data channels (similar to the pipes model in Unix). These channels are basically asynchronous FIFO queues. Any process can define one or more channels as input and output. The order of interaction between these processes, and ultimately the order of workflow execution depends on the communication channel dependencies between processes. For example, if process A writes data to channel A and process B reads data from channel A, then Nextflow knows that process A must be executed before process B.
+
+Check this source code example: [seq_ocrd_wf_many.nf](https://github.com/MehmedGIT/OPERANDI_TestRepo/blob/master/ExampleWorkflows/Nextflow/workflow4/seq_ocrd_wf_many.nf)
+
+TODO: I will provide more parallelization details here based on the example above.
+
+## How does the NF script interact with the processing server?
+There is still no running processing server. More details will be announced once there is more to talk about. The interaction will most probably happen with curl through a bash script inside the Nextflow process. Of course, if it is integrated inside the OCR-D core, then no direct interactions will be needed from inside the Nextflow script.
+
+## How does the NF script interact with the METS server?
+There is still no running METS server. More details will be announced once there is more to talk about. The interaction will most probably happen with curl through a bash script inside the Nextflow process. Of course, if it is integrated inside the OCR-D core, then no direct interactions will be needed from inside the Nextflow script.
+
+## How to convert the existing OCR-D process workflows we reference to NF?
+I have written an OtoN (OCR-D to Nextflow) converter which converts basic OCR-D process workflows to Nextflow workflow scripts. Check here: [OtoN](https://github.com/MehmedGIT/OtoN_Converter)

```

Comment:
You are right. I have to change all occurrences of "I" to "we". It is also academically more correct to do it that way. Just like in research papers.

Reply:
Yes, "we" is better. Also regarding conciseness, my questions weren't supposed to be exhaustive or precise, should be more a guide to the document, rather than a FAQ. But we can improve this subsequently.